### PR TITLE
Handle null tags (the default) in effective-services-mojo

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please review and merge.

### CODE REVIEW

File: vespa-maven-plugin/src/main/java/ai/vespa/hosted/plugin/EffectiveServicesMojo.java [hunk 1]
```diff
@@ -45,11 +45,12 @@ protected void doExecute() throws Exception {
...
         ZoneId zone = zoneOf(environment, region);
         Path output = Paths.get(outputDirectory).resolve("services-" + zone.environment().value() + "-" + zone.region().value() + ".xml");
-        Files.write(output, effectiveServices(services, zone, InstanceName.from(instance), Tags.fromString(tags)).getBytes(StandardCharsets.UTF_8));
+        Tags tagz = Tags.fromString(tags);
+        Files.write(output, effectiveServices(services, zone, InstanceName.from(instance), tagz).getBytes(StandardCharsets.UTF_8));
         getLog().info("Effective services for " + zone +
                       ", instance " + instance +
-                      ( tags.isEmpty() ? "" : ", tags " + tags ) +
-                      " written to " + output);
+                      ( tags == null ? "" : ", tags '" + tagz) +
```
Comment (jonmv) [inline vespa-maven-plugin/src/main/java/ai/vespa/hosted/plugin/EffectiveServicesMojo.java:52]: Showing actual tags if the argument was specified.

Comment (jonmv) [reply vespa-maven-plugin/src/main/java/ai/vespa/hosted/plugin/EffectiveServicesMojo.java:52]: Urgh, got the `'` wrong.

File: vespa-maven-plugin/src/main/java/ai/vespa/hosted/plugin/EffectiveServicesMojo.java [hunk 2]
```diff
@@ -45,11 +45,12 @@ protected void doExecute() throws Exception {
...
+                      "' written to " + output);
```
Comment (jonmv) [inline vespa-maven-plugin/src/main/java/ai/vespa/hosted/plugin/EffectiveServicesMojo.java:53]:
```suggestion
                      ( tags == null ? "" : ", tags '" + tagz + "' ") +
                      "written to " + output);
```

File: vespa-maven-plugin/src/main/java/ai/vespa/hosted/plugin/EffectiveServicesMojo.java [hunk 3]
```diff
@@ -45,11 +45,12 @@ protected void doExecute() throws Exception {
...
         Path output = Paths.get(outputDirectory).resolve("services-" + zone.environment().value() + "-" + zone.region().value() + ".xml");
-        Files.write(output, effectiveServices(services, zone, InstanceName.from(instance), Tags.fromString(tags)).getBytes(StandardCharsets.UTF_8));
+        Tags tagz = Tags.fromString(tags);
+        Files.write(output, effectiveServices(services, zone, InstanceName.from(instance), tagz).getBytes(StandardCharsets.UTF_8));
         getLog().info("Effective services for " + zone +
                       ", instance " + instance +
-                      ( tags.isEmpty() ? "" : ", tags " + tags ) +
-                      " written to " + output);
+                      ( tags == null ? "" : ", tags '" + tagz + "' ") +
+                      "written to " + output);
```
Comment (jonmv) [inline vespa-maven-plugin/src/main/java/ai/vespa/hosted/plugin/EffectiveServicesMojo.java:53]:
```suggestion
                      ( tags == null ? "" : ", tags '" + tagz + "'") +
                      " written to " + output);
```

**Review by bratseth — APPROVED** [review]

**Pull request merged**